crates/types/protobuf/restate/common.proto

```diff
@@ -35,6 +35,23 @@ message Lsn { uint64 value = 1; }
 // A generic type for versioned metadata
 message Version { uint32 value = 1; }
 
+// Represents the API versioning information.
+message ApiVersion {
+  // The minimum API version supported.
+  // This should only be set when introducing a breaking change,
+  // such as modifying an existing request message.
+  // Adding new API handlers does not require updating this value,
+  // as older clients remain compatible with new services.
```

Comment:
One of the benefits of using grpc is to try and be as backward compatible as possible. It's uncommon to version grpc services this way.

One strategy is to add the version to the grpc package name, for instance `restate.cluster.v1` in `restate.cluster.v1.ClusterCtlSvc`, albeit I would defer introducing a `v1` prefix until it's necessarily (at the first incompatible API shape). This means it'd be possible to maintain two versions concurrently and the client chooses.

Reply:
This is not about versioning though, rather it's about version-awareness. The strategies for versioning and backwards-compatible evolution are somewhat orthogonal to the purpose (in my mind!) of propagating the server version info back to the caller. That is (to me) is to be able to warn about mismatches, which could have semantic gaps and thus lead to operator errors, even if the wire messages are perfectly compatible.